examples/mem-performance-test/src/main.rs

```diff
@@ -0,0 +1,270 @@
+        if vec.push_within_capacity(mem).is_err() {
+            break;
+        }
+        // vec.push(&mut mem);
+        i += 1;
+        if i >= n_els { 
+            break; 
+        }
+    }
+    total_ops = i;
+
+    // *** Free *** //
+    for _ in 0..total_ops {
+        if let Some(mem) = vec.pop() {
+            drop(mem);
+        }
+    }
+    let end = Instant::now();
+
+    // *** now we can remove the elememts from the vector *** /
+    vec.clear();
+
+    debug_assert_ne!(total_ops, 0);
+    let total_duration = end - start;
+    assert!(total_duration.as_secs_f64() > 0.0);
+
+    // return operations per second
+    Ok((total_duration, total_ops as u64))
+}
+
+// if n_sec == 0, do a single run (ignore_first is ignored)
+#[allow(dead_code)]
+fn test_alloc_free_via_vector(buf_size: usize, n_sec: u32, ignore_first: bool) 
+                              -> Result<f64, AllocError> {
+    debug_assert_ne!(buf_size, 0);
+
+    // lets start with the default max data size addressable by vec, if not set already
+    let n_els = DEFAULT_MAX_MEM_ALLOC_SIZE / buf_size;
+    debug_assert_ne!(n_els, 0);
+    let mut vec: Vec<Box<[u64]>> = Vec::with_capacity(n_els);
+
+    let mut total_duration = Duration::default();
+    let mut total_operations: u64 = 0;
+    let mut first = true;
+    let start = Instant::now();
+    loop {
+        match alloc_free_via_vector(&mut vec, buf_size, DEFAULT_MAX_MEM_ALLOC_SIZE) {
+            Ok((dur, oper)) => {
+                debug_assert_ne!(oper, 0);
+                if n_sec == 0 {
+                    return Ok(oper as f64 / dur.as_secs_f64());
+                }
+                if ignore_first && first {
+                    first = false;
+                } else {
+                    total_duration += dur;
+                    total_operations += oper;
+                }
+            },
+            Err(err) => {
+                println!("alloc_free_via_vector failed with {err}");
+                return Err(err);
+            },
+        }
+        let dur = Instant::now() - start;
+        if dur.as_secs_f64() as u64 > n_sec as u64 {
+            break;
+        }
+    }
+
+    return Ok(total_operations as f64 / total_duration.as_secs_f64());
+}
+
+fn run_tests(size: usize, test_sec: u32) -> Result<f64, AllocError> {
+    #[cfg(feature = "mem_hot")]
+        return test_alloc_free_via_vector(size, test_sec, false);
+
+    #[cfg(feature = "mem_cold")]
+        return test_alloc_free_via_vector(size, test_sec, true);
+
+    return test_alloc_free(size, test_sec);
+}
+
+fn get_test_name() -> &'static str {
+    #[cfg(feature = "mem_hot")]
+        return "Memory Hot Tests";
+
+    #[cfg(feature = "mem_cold")]
+        return "Memory Cold Tests (single pass)";
+
+    return "Basic Memory tests";
+}
+
+fn get_zero_after_alloc_txt() -> &'static str {
+    #[cfg(feature = "zero_after_alloc")]
+        return "zero_after_alloc";
+
+    return "!zero_after_alloc";
+}
+
+fn main() {
```

Comment:
Can you add some of the test results as a file in the `examples/mem-perforamnce-test` directory so someone who runs the test in the future is aware of expected results?